### PR TITLE
Add playbooks of IBM® Cloud Infrastructure Center OCP UPI

### PR DESCRIPTION
**Pull request by gyohuangxin**

### CODE REVIEW

**Comment by ddimatos** [comment]
@gyohuangxin - Hello and thank you for this, I will be working with the team and getting back to you hopefully this week.

**Comment by gyohuangxin** [comment]
> @gyohuangxin - Hello and thank you for this, I will be working with the team and getting back to you hopefully this week.

Thanks! no worries, that sounds good.

**Comment by zhengxiaomei123** [comment]
Hi @ddimatos , Is it possible to review this PR this week? We have some clients who wait for the resource. Thanks very much.

File: cloud_infra_center/ocp_upi/add-new-compute-node.yaml
Comment (ddimatos) [inline cloud_infra_center/ocp_upi/add-new-compute-node.yaml:2]: I will check with our legal team but I do believe at that IBM surrenders ownership by use of Apache 2.0 License. I will get back to you on this.

File: cloud_infra_center/ocp_upi/common.yaml
Comment (ddimatos) [inline cloud_infra_center/ocp_upi/common.yaml:2]: You have a URL going out to another repo here after the copyright. "https://github.com/openshift/installer"

File: cloud_infra_center/ocp_upi/README.md
Comment (ddimatos) [inline cloud_infra_center/ocp_upi/README.md:15]: These indentations in Markdown don't render correctly as bullets when you look at the README

File: cloud_infra_center/ocp_upi/README.md
Comment (ddimatos) [inline cloud_infra_center/ocp_upi/README.md:21]: Bullets 1) - 3) are on one single line, you probably want to them as single lines

File: cloud_infra_center/ocp_upi/README.md
Comment (itsBryantP) [inline cloud_infra_center/ocp_upi/README.md:46]: typo woker -> worker

File: cloud_infra_center/ocp_upi/inventory.yaml
Comment (ddimatos) [inline cloud_infra_center/ocp_upi/inventory.yaml:68]: I wonder why you chose to have hard coded IPs here instead of `{{ vars }}`

File: cloud_infra_center/ocp_upi/roles/configure-installer-and-image/tasks/main.yaml
Comment (itsBryantP) [inline cloud_infra_center/ocp_upi/roles/configure-installer-and-image/tasks/main.yaml:16]: suggest removing hardcoded URLs to a variables file that can be easily changed should the URLs change in the future.

**Review by ddimatos — APPROVED** [review]
I have never tried to execute a python script from a role, one concern I have is the permission bits might need to be corrected after a user clones the repo or copies the content and if you have tested the permission bit requirements and documented it.

File: cloud_infra_center/ocp_upi/roles/configure-security-groups/tasks/main.yaml
Comment (itsBryantP) [inline cloud_infra_center/ocp_upi/roles/configure-security-groups/tasks/main.yaml:52]: suggest moving these ports to a defaults variables file or even higher level so they are not hardcoded in the role itself.

**Comment by ddimatos** [comment]
Hi, we made a few comments, I will have more but from what I could see none of the comments justified holding this PR from merging and given you have a customer wanting to access it, we feel its ok to merge now. I will ask that you review some of the comments, I will have more in time. 

Could you provide me two people who will maintain this playbook?
- Name:
- Email:
- git ID/handle:

File: cloud_infra_center/ocp_upi/common.yaml
Comment (gyohuangxin) [reply cloud_infra_center/ocp_upi/common.yaml:2]: Yes, because this yaml file was copied from OpenShift repo, I was told to keep the same copyright for legal reason.

File: cloud_infra_center/ocp_upi/README.md
Comment (gyohuangxin) [reply cloud_infra_center/ocp_upi/README.md:15]: Thanks, I'll record and fix in another PR.

File: cloud_infra_center/ocp_upi/README.md
Comment (gyohuangxin) [reply cloud_infra_center/ocp_upi/README.md:46]: Thanks, will fix in another PR.

File: cloud_infra_center/ocp_upi/README.md
Comment (gyohuangxin) [reply cloud_infra_center/ocp_upi/README.md:21]: Will fix.

File: cloud_infra_center/ocp_upi/inventory.yaml
Comment (gyohuangxin) [reply cloud_infra_center/ocp_upi/inventory.yaml:68]: The reason why hardcoded is that the items in inventory.yaml was an example for users. It's a good suggestion any how and I'll consider in the future.

File: cloud_infra_center/ocp_upi/roles/configure-security-groups/tasks/main.yaml
Comment (gyohuangxin) [reply cloud_infra_center/ocp_upi/roles/configure-security-groups/tasks/main.yaml:52]: Good suggestion, and I will consider in the future.

**Comment by gyohuangxin** [comment]
@ddimatos @itsBryantP Thank you so much! Your comments are very helpful for us, and we will consider them in the future PRs.
And thanks for adding us as maintainers, both of you can leave any comments on our code at any time!

File: cloud_infra_center/ocp_upi/common.yaml
Comment (ddimatos) [reply cloud_infra_center/ocp_upi/common.yaml:2]: That makes sense, you are correct that we need to honor other parties copyrights , good thing is its the same License, but I still want to check with Legal to see if this is enough to have a link or do we need more.

Comment (gyohuangxin) [reply cloud_infra_center/ocp_upi/common.yaml:2]: Thanks, sounds good~

**Pull request merged**
